### PR TITLE
Merge v2.0.23 and v2.0.24 into master.

### PR DESCRIPTION
**Pull request by SaberUK**

### CODE REVIEW

File: src/modules/m_permchannels.cpp [hunk 1]
```diff
@@ -221,6 +221,12 @@ class ModulePermanentChannels : public Module
 
 				ServerInstance->Logs->Log(MODNAME, LOG_DEBUG, "Added %s with topic %s", channel.c_str(), c->topic.c_str());
 
+				if (modes.find('P') == std::string::npos)
+				{
+					ServerInstance->Logs->Log("m_permchannels", LOG_DEFAULT, "%s (%s) does not have +P set in <permchannels:modes>; it will be deleted when empty!",
+						c->name.c_str(), tag->getTagLocation().c_str());
+				}
+
```
Comment (SISheogorath) [inline src/modules/m_permchannels.cpp]: If I recall correct @attilamolnar said we should use a mode handler for this(?)

File: src/modules/m_sasl.cpp [hunk 2]
```diff
@@ -378,16 +435,14 @@ class ModuleSASL : public Module
 		servertracker.Reset();
 	}
 
-	ModResult OnUserRegister(LocalUser *user) CXX11_OVERRIDE
+	void OnUserConnect(LocalUser *user)
```
Comment (SISheogorath) [inline src/modules/m_sasl.cpp]: No need for `CXX11_OVERRIDE` here?

File: src/modules/m_cgiirc.cpp [hunk 3]
```diff
@@ -85,6 +86,14 @@ class CommandWebirc : public Command
 			if(user->registered == REG_ALL)
 				return CMD_FAILURE;
 
+			irc::sockets::sockaddrs ipaddr;
+			if (!irc::sockets::aptosa(parameters[3], 0, ipaddr))
+			{
+				IS_LOCAL(user)->CommandFloodPenalty += 5000;
```
Comment (SISheogorath) [inline src/modules/m_cgiirc.cpp]: Mhm okay, looks like I mixed up `IS_LOCAL` with `IS_OPER`

File: src/modules/m_cgiirc.cpp [hunk 4]
```diff
@@ -112,6 +121,7 @@ class CommandWebirc : public Command
 				}
 			}
 
+			IS_LOCAL(user)->CommandFloodPenalty += 5000;
```
Comment (SISheogorath) [inline src/modules/m_cgiirc.cpp]: Mhm okay, looks like I mixed up `IS_LOCAL` with `IS_OPER`

File: src/modules/m_permchannels.cpp (hunk 1 above)
Comment (SaberUK) [inline src/modules/m_permchannels.cpp]: Yeah, theres a PR open for that already. When this gets merged i'll remove the notice in the other PR.

File: src/modules/m_sasl.cpp (hunk 2 above)
Comment (SaberUK) [inline src/modules/m_sasl.cpp]: Good catch, i'll add that.

**Comment by rburchell** [comment]
Looks good, thanks!

File: src/modules/m_permchannels.cpp (hunk 1 above)
Comment (SaberUK) [inline src/modules/m_permchannels.cpp]: The other one got merged so I removed it in this.

**Pull request merged**